app/controllers/api/v1x1.rb

```diff
@@ -6,17 +6,52 @@ def openapi
       end
     end
 
-    class ApprovalRequestsController          < Api::V1x0::ApprovalRequestsController; end
-    class GraphqlController                   < Api::V1x0::GraphqlController; end
-    class OrderItemsController                < Api::V1x0::OrderItemsController; end
-    class OrdersController                    < Api::V1x0::OrdersController; end
-    class PortfolioItemsController            < Api::V1x0::PortfolioItemsController; end
-    class PortfoliosController                < Api::V1x0::PortfoliosController; end
-    class ProgressMessagesController          < Api::V1x0::ProgressMessagesController; end
-    class ProviderControlParametersController < Api::V1x0::ProviderControlParametersController; end
-    class ServicePlansController              < Api::V1x0::ServicePlansController; end
-    class SettingsController                  < Api::V1x0::SettingsController; end
-    class TagsController                      < Api::V1x0::TagsController; end
-    class TenantsController                   < Api::V1x0::TenantsController; end
+    class ApprovalRequestsController          < Api::V1x0::ApprovalRequestsController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class GraphqlController                   < Api::V1x0::GraphqlController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class OrderItemsController                < Api::V1x0::OrderItemsController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class OrdersController                    < Api::V1x0::OrdersController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class PortfolioItemsController            < Api::V1x0::PortfolioItemsController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class PortfoliosController                < Api::V1x0::PortfoliosController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class ProgressMessagesController          < Api::V1x0::ProgressMessagesController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class ProviderControlParametersController < Api::V1x0::ProviderControlParametersController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class ServicePlansController              < Api::V1x0::ServicePlansController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class SettingsController                  < Api::V1x0::SettingsController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class TagsController                      < Api::V1x0::TagsController
+      include Api::V1x1::Mixins::IndexMixin
+    end
+
+    class TenantsController                   < Api::V1x0::TenantsController
+      include Api::V1x1::Mixins::IndexMixin
+    end
```

Comment:
This feels kinda gross. I think we can do something similar to how I originally was going to introduce versioning in [this PR](https://github.com/RedHatInsights/catalog-api/pull/626/files).

Basically, we have `config/initializers/module.rb` that extends module with the `extend_each_subclass` method [here](https://github.com/RedHatInsights/catalog-api/pull/626/files#diff-5c19dd8dbae15938a929a9264f1f7abd), and then we do something similar to [this](https://github.com/RedHatInsights/catalog-api/pull/626/files#diff-2aa2a4b72cfc6cfcb6c2542db5fb91e1R23) for the actual index mixin includes.

Thoughts?

Reply:
Yep I'm down for that, looks MUCH cleaner that way, and it makes doing this type of stuff much easier for future versions.